File: .claude/hooks/log_action.py
#!/usr/bin/env python3
"""PreToolUse/PostToolUse: registra cada acción del agente en ops/actions/<run_id>.jsonl."""
import json
import os
import sys
from datetime import datetime, timezone
from pathlib import Path

MAX = 300


def main() -> None:
    run_id = os.environ.get("FACTORY_RUN_ID")
    if not run_id:
        sys.exit(0)
    try:
        event = json.load(sys.stdin)
    except json.JSONDecodeError:
        sys.exit(0)

    out_dir = Path(os.environ.get("FACTORY_REPO", ".")) / "ops" / "actions"
    out_dir.mkdir(parents=True, exist_ok=True)
    record = {
        "ts": datetime.now(timezone.utc).isoformat(timespec="seconds"),
        "run_id": run_id,
        "agent": os.environ.get("FACTORY_AGENT"),
        "event": event.get("hook_event_name"),
        "tool": event.get("tool_name"),
        "input": json.dumps(event.get("tool_input") or {}, ensure_ascii=False)[:MAX],
    }
    with open(out_dir / f"{run_id}.jsonl", "a", encoding="utf-8") as fh:
        fh.write(json.dumps(record, ensure_ascii=False) + "\n")
    sys.exit(0)


if __name__ == "__main__":
    main()

File: .claude/hooks/log_action.ts
/** PreToolUse/PostToolUse: registra cada acción del agente en ops/actions/<run_id>.jsonl. */
import { appendFileSync, mkdirSync, readFileSync } from "node:fs";
import { join } from "node:path";

const MAX = 300;

function main(): void {
  const runId = process.env.FACTORY_RUN_ID;
  if (!runId) return;
  let event;
  try {
    event = JSON.parse(readFileSync(0, "utf8"));
  } catch {
    return;
  }

  const outDir = process.env.FACTORY_LOG_DIR || join(process.env.FACTORY_REPO || ".", "ops", "actions");
  mkdirSync(outDir, { recursive: true });
  const record = {
    ts: new Date().toISOString().slice(0, 19) + "+00:00",
    run_id: runId,
    agent: process.env.FACTORY_AGENT,
    event: event.hook_event_name,
    tool: event.tool_name,
    input: JSON.stringify(event.tool_input ?? {}).slice(0, MAX),
  };
  appendFileSync(join(outDir, `${runId}.jsonl`), JSON.stringify(record) + "\n", "utf8");
}

main();
